src/CMakeLists.txt

```diff
@@ -1,17 +1,12 @@
-file(GLOB SOURCE_FILES *.cpp expression/*.cpp frame_selection/*.cpp ontology/*.cpp)
+file(GLOB SOURCE_FILES *.cpp *.h */*.h */*.cpp */*.c)
 
 include_directories(../libs/pstream)
 include_directories(../libs/fast-cpp-csv-parser) # May need to compile with thread control
 
-include_directories(database)
-link_directories(database)
-add_subdirectory(database)
-
 include_directories(${Boost_INCLUDE_DIRS})
 add_executable(EnTAP ${SOURCE_FILES})
 
-
-target_link_libraries(EnTAP boost pthread EntapDatabase SQLDatabaseHelper sqlite3)
+target_link_libraries(EnTAP dl pthread Boost::program_options Boost::system Boost::filesystem Boost::iostreams Boost::serialization Boost::regex Boost::date_time)
```

Comment:
Why the change here from 'boost' to the individual library callouts(or ${Boost_LIBRARIES})? Did that help your linking issues.